mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -59,6 +59,166 @@ struct RockLinalgAlignPass
   void runOnOperation() override;
 };
 
+// This rewrite will rewrite the linalg IO that has view like-ops surrounding
+// them to be consumed by the linalg operation itself adjusting the indexing
+// maps to faithfully represent them.
+struct InlineViewLikeOperandsLinalgRewritePattern
+    : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp laGeneric,
+                                PatternRewriter &rewriter) const override;
+};
+
+// This function will take strides calculated using reassociation of dimensions
+// that are modified by view-like ops and recompute the affine expr from src
+// view to dest view.
+llvm::SmallVector<mlir::AffineExpr, 4>
```

Comment:
Nit: You can take in `SmallVectorImpl<AffineExpr> &out` here and save on copies in the loop below.

Also, neither SmallVector nor AffineExpr need namespaces on them, I think

Reply:
Done